test/Plugins.Tests/SharedTypesTests.cs

```diff
@@ -47,11 +47,18 @@ public void TransitiveAssembliesOfSharedTypesAreResolved()
                     TestResources.GetTestProjectAssembly("TransitivePlugin"),
                     sharedTypes: new[] { typeof(SharedType) });
 
-            var assembly = loader.LoadDefaultAssembly();
-            var configType = assembly.GetType("TransitivePlugin.PluginConfig", throwOnError: true)!;
-            var config = Activator.CreateInstance(configType);
-            var transitiveInstance = configType.GetMethod("GetTransitiveType")?.Invoke(config, null);
-            Assert.IsType<Test.Transitive.TransitiveSharedType>(transitiveInstance);
+            TransitiveAssembliesOfSharedTypesAreResolvedBase(loader);
+        }
+
+        /// <summary>
+        /// Variant of <see cref="TransitiveAssembliesOfSharedTypesAreResolved"/> which uses lazy loading for assemblies.
+        /// </summary>
+        [Fact]
```

Comment:
Instead of two separate test methods, this could be simplified by just making IsLazyLoaded a parameter.
```c#
[Theory]
[InlineData(true)]
[InlineData(false)]
public void TransitiveAssembliesOfSharedTypesAreResolved(bool isLazyLoaded)
```

Reply:
Completely agree here; not really sure what I was thinking.
Maybe I thought that it would be easier to look at when checking reports. 
That was silly of me; I'll fix that.